src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppAppRunner/Generated/Recipe.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            if (ServiceAccessRole == null)
+                throw new InvalidOperationException($"{nameof(ServiceAccessRole)} has not been set. The {nameof(ConfigureIAMRoles)} method should be called before {nameof(ConfigureAppRunnerService)}");
+            if (TaskRole == null)
+                throw new InvalidOperationException($"{nameof(TaskRole)} has not been set. The {nameof(ConfigureIAMRoles)} method should be called before {nameof(ConfigureAppRunnerService)}");
+
+            if (string.IsNullOrEmpty(props.ECRRepositoryName))
+                throw new InvalidOrMissingConfigurationException("The provided ECR Repository Name is null or empty.");
+
+            var ecrRepository = Repository.FromRepositoryName(this, "ECRRepository", props.ECRRepositoryName);
+
+            Configuration settings = props.Settings;
+            var appRunnerServiceProp = new CfnServiceProps
+            {
+                ServiceName = settings.ServiceName,
+                SourceConfiguration = new CfnService.SourceConfigurationProperty
+                {
+                    AuthenticationConfiguration = new CfnService.AuthenticationConfigurationProperty
+                    {
+                        AccessRoleArn = ServiceAccessRole.RoleArn
+                    },
+                    ImageRepository = new CfnService.ImageRepositoryProperty
+                    {
+                        ImageRepositoryType = "ECR",
+                        ImageIdentifier = ContainerImage.FromEcrRepository(ecrRepository, props.ECRImageTag).ImageName,
+                        ImageConfiguration = new CfnService.ImageConfigurationProperty
+                        {
+                            Port = settings.Port.ToString(),
+                            StartCommand = !string.IsNullOrWhiteSpace(settings.StartCommand) ? settings.StartCommand : null
+                        }
+                    }
+                }
+            };
+
+            if (!string.IsNullOrEmpty(settings.EncryptionKmsKey))
+            {
+                var encryptionConfig = new CfnService.EncryptionConfigurationProperty();
+                appRunnerServiceProp.EncryptionConfiguration = encryptionConfig;
+
+                encryptionConfig.KmsKey = settings.EncryptionKmsKey;
+            }
+
+            var healthCheckConfig = new CfnService.HealthCheckConfigurationProperty();
+            appRunnerServiceProp.HealthCheckConfiguration = healthCheckConfig;
+
+            healthCheckConfig.HealthyThreshold = settings.HealthCheckHealthyThreshold;
+            healthCheckConfig.Interval = settings.HealthCheckInterval;
+            healthCheckConfig.Protocol = settings.HealthCheckProtocol;
+            healthCheckConfig.Timeout = settings.HealthCheckTimeout;
+            healthCheckConfig.UnhealthyThreshold = settings.HealthCheckUnhealthyThreshold;
+
+            if (string.Equals(healthCheckConfig.Protocol, "HTTP"))
```

Comment:
nit: is there a constant that we can reference instead of using a string "HTTP"? or use StringComparison.OrdinalIgnoreCase ?